recipes/security_init.rb

```diff
@@ -39,3 +39,26 @@
   not_if { File.exist?(path) }
   only_if { ssl_enabled }
 end
+
+# Manage Authentication realmfile
+if node['cdap']['security']['manage_realmfile'].to_s == 'true'
+  if node.key?('cdap') && node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('security.authentication.handlerClassName') &&
+     node['cdap']['cdap_site']['security.authentication.handlerClassName'] == 'co.cask.cdap.security.server.BasicAuthenticationHandler'
+    if node.key?('cdap') && node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('security.authentication.basic.realmfile')
+      realmfile = node['cdap']['cdap_site']['security.authentication.basic.realmfile']
+      realmdir = ::File.dirname(realmfile)
+
+      # Ensure parent directory exists
+      directory realmdir do
+        action :create
+        recursive true
+      end
+
+      # Create the realmfile
+      file realmfile do
+        content "#{node['cdap']['security']['realmfile']['username']}: #{node['cdap']['security']['realmfile']['password']}"
```

Comment:
i hope nobody is using it for more than testing, but will add a attribute-driven template instead